web/js/header.js

```diff
@@ -1,13 +1,26 @@
-var headerModule = (function () {
+let headerModule = (function () {
     return {
         toCreateContact: function () {
             window.location = "http://localhost:8080/contactManager?command=redirect&page=addContact";
         },
-        toContactList: function () {
-            window.location = "http://localhost:8080/contactManager?command=redirect&page=contactList";
+        toContactList: function (error) {
+            if (error !== null) {
+                window.location = `http://localhost:8080/contactManager?command=redirect&page=contactList&error=${error}`;
+            } else {
+                window.location = `http://localhost:8080/contactManager?command=redirect&page=contactList`;
+            }
         },
-        toPhoneNumbersList:function () {
-            window.location="http://localhost:8080/contactManager?command=redirect&page=phoneNumberList"
+        sendEmail: function () {
+            let contactTable = document.getElementById('allContactsTable');
+            if (contactTable !== null) {
```

Comment:
It would be look better: `contactTable ? ShowAllContactsCommand().execute(request, response) : gatherEmails()`